src/sample/db/database.sql

```diff
@@ -0,0 +1,87 @@
+CREATE DATABASE COURSE_REGISTER;
+USE COURSE_REGISTER;
+
+CREATE TABLE COURSE(
+ COURSE_ID CHAR(6) NOT NULL,
+ COURSE_NAME VARCHAR(100) NOT NULL,
+ PRIMARY KEY(COURSE_ID)
+ );
+ 
+ CREATE TABLE STUDENT(
+  MATRIC_NUMBER CHAR(6) NOT NULL,
+  STU_NAME VARCHAR(30) NOT NULL,
+  STU_IC_NUM CHAR(5) NOT NULL,
+  ACADEMIC_STATUS VARCHAR(8) NOT NULL,
+  SEMESTER_REGISTERED CHAR(3) NOT NULL,
+  STU_CGPA DECIMAL(3,2) NOT NULL,
+  STU_YEAR CHAR(1) NOT NULL,
+  STU_SCHOOL VARCHAR(50) NOT NULL,
+  STU_CAMPUS VARCHAR(50) NOT NULL,
+  STU_PROGRAMME VARCHAR (50) NOT NULL,
+  PRIMARY KEY(MATRIC_NUMBER)
+  );
+  
+  CREATE TABLE LECTURER(
+   STAFF_ID CHAR(6) NOT NULL,
+   LECT_NAME VARCHAR(30) NOT NULL,
+   LECT_IC_NUMBER CHAR(5) NOT NULL,
+   LECT_STATUS VARCHAR(8) NOT NULL,
+   LECT_SCHOOL VARCHAR(50) NOT NULL,
+   LECT_POSITION VARCHAR(50) NOT NULL,
+   PRIMARY KEY(STAFF_ID)
+   );
+ 
+
+   
+CREATE TABLE STUDENT_ACCOUNT(
+ STU_EMAIL VARCHAR(319) NOT NULL,
+ STU_PASSWORD VARCHAR(100) NOT NULL,
+ MATRIC_NUMBER CHAR(6) NOT NULL,
+ PRIMARY KEY (STU_EMAIL),
+ FOREIGN KEY(MATRIC_NUMBER) 
+	REFERENCES STUDENT (MATRIC_NUMBER) ON DELETE CASCADE
+ );
+ 
+ CREATE TABLE COURSE_STUDENT_LECT(
+  COURSE_ID CHAR(6) NOT NULL,
+  MATRIC_NUMBER CHAR(6) NOT NULL,
+  STAFF_ID CHAR(6) NOT NULL,
+  STUDENT_COURSE_STATUS CHAR(1) NOT NULL,
```

Comment:
What is student_coirse_status for?

Reply:
It's used by us to track whether the students have taken the course or not in order to recommend the courses for them to register in that particular semester.

Reply:
why need student_course_status, if the students did not register for that course, just no need to have a record for that student is enough right?

Reply:
the recommendation algorithm would be like as follow:
SELECT COURSE_RECOMMENDATION FROM STUDENT_COURSE_STATUS WHERE COURSE_STATUS is 'y';

Reply:
Then if a student did not take that course, it will automatically be recommended? If this is the case, why bother to have course recommendation?

Reply:
That's just an example of showing the function of student_course_status. When referring to database we don't know whether the student has taken the course or not. So via the status , we can then recommend the course that is not taken by them. If the status does not exist, then we can't know what are the courses supposed to take by them and not taken by them. We then can't recommend the courses to them.

Reply:
But then I think I can just remove it if it's unnecessary. Cuz I think we shall proceed to the controller and connection to see whether this database is okay anot. This database might not the most perfect one but we can work on it incrementally.